File: 665-non-decreasing-array/665-non-decreasing-array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var checkPossibility = function(nums) {
    let cnt = 0;
    
    for (let i = 1; i < nums.length; i++) {
        if (nums[i - 1] <= nums[i]) continue;
        if (i > 1 && nums[i - 2] > nums[i]) nums[i] = nums[i - 1];
        
        cnt++;
    }
    
    return cnt < 2;
};

File: 665-non-decreasing-array/665-non-decreasing-array.py
class Solution:
    def checkPossibility(self, nums: List[int]) -> bool:
        count = 0
        for i in range(1, len(nums)):
            if nums[i - 1] > nums[i]:
                count += 1
                if i > 1 and nums[i - 2] > nums[i]:
                    nums[i] = nums[i - 1]
                
                
        return count < 2